src/main/java/frc/robot/commands/IntakeGamePiece.java

```diff
@@ -6,30 +6,41 @@
 
 import edu.wpi.first.wpilibj2.command.Commands;
 import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import frc.robot.Constants.GamePiece;
 import frc.robot.RobotPreferences.prefElevator;
 import frc.robot.RobotPreferences.prefIntake;
 import frc.robot.RobotPreferences.prefWrist;
 import frc.robot.subsystems.Elevator;
 import frc.robot.subsystems.Intake;
 import frc.robot.subsystems.Wrist;
 
-public class IntakeCone extends SequentialCommandGroup {
-
+public class IntakeGamePiece extends SequentialCommandGroup {
   Wrist subWrist;
   Intake subIntake;
   Elevator subElevator;
+  GamePiece gamepiece;
+
+  double intakeSpeed;
 
-  public IntakeCone(Wrist subWrist, Intake subIntake, Elevator subElevator) {
+  public IntakeGamePiece(Wrist subWrist, Intake subIntake, Elevator subElevator, GamePiece gamepiece) {
 
     this.subWrist = subWrist;
     this.subIntake = subIntake;
+    this.subElevator = subElevator;
+    this.gamepiece = gamepiece;
+
+    if (gamepiece == GamePiece.CONE) {
+      intakeSpeed = prefIntake.intakeConeSpeed.getValue();
+    } else if (gamepiece == GamePiece.CUBE) {
+      intakeSpeed = prefIntake.intakeCubeSpeed.getValue();
+    }
```

Comment:
Could you add a fallback condition? I know it should never be anything else but redundancy doesn't hurt